python-wrapper/src/neo4j_viz/gql_create.py

```diff
@@ -0,0 +1,337 @@
+import re
+import uuid
+from typing import Any, Optional
+
+from neo4j_viz import Node, Relationship, VisualizationGraph
+
+
+def _parse_value(value_str: str) -> Any:
+    value_str = value_str.strip()
+    if not value_str:
+        return None
+
+    # Parse object
+    if value_str.startswith("{") and value_str.endswith("}"):
+        inner = value_str[1:-1].strip()
+        result = {}
+        depth = 0
+        in_string = None
+        start_idx = 0
+        for i, ch in enumerate(inner):
+            if in_string is None:
+                if ch in ["'", '"']:
+                    in_string = ch
+                elif ch in ["{", "["]:
+                    depth += 1
+                elif ch in ["}", "]"]:
+                    depth -= 1
+                elif ch == "," and depth == 0:
+                    segment = inner[start_idx:i].strip()
+                    if ":" not in segment:
+                        return None
+                    k, v = segment.split(":", 1)
+                    k = k.strip().strip("'\"")
+                    result[k] = _parse_value(v)
+                    start_idx = i + 1
+            else:
+                if ch == in_string:
+                    in_string = None
+        if inner[start_idx:]:
```

Comment:
should we very the depth is equal to 0 at the end here?

Reply:
We don't have to since we already do it below (as you also note)